spec/safe_statements_spec.rb

```diff
@@ -3484,6 +3496,95 @@ def up
               end
             end
 
+            it "does not wait to acquire a lock if same table in different schema is blocked" do
+              stub_const("PgHaMigrations::LOCK_TIMEOUT_SECONDS", 1)
+
+              ActiveRecord::Base.connection.execute("CREATE TABLE partman.bogus_table(pk SERIAL, i INTEGER)")
+
+              begin
+                thread = Thread.new do
+                  migration.safely_acquire_lock_for_table("partman.bogus_table") { sleep 2 }
+                end
+
+                sleep 1.1
+
+                expect(PgHaMigrations::BlockingDatabaseTransactions).to receive(:find_blocking_transactions)
+                  .once
+                  .and_call_original
+
+                migration.suppress_messages do
+                  migration.safely_acquire_lock_for_table(table_name) do
+                    expect(locks_for_table("partman.bogus_table", connection: alternate_connection)).not_to be_empty
+                    expect(locks_for_table(table_name, connection: alternate_connection)).not_to be_empty
+                  end
+                end
+              ensure
+                thread.join
+              end
+            end
+
+            it "waits to acquire a lock if the table is partitioned and child table is blocked" do
+              stub_const("PgHaMigrations::LOCK_TIMEOUT_SECONDS", 1)
+
+              ActiveRecord::Base.connection.drop_table(table_name)
+              create_range_partitioned_table(table_name, migration_klass, with_partman: true)
+
+              begin
+                thread = Thread.new do
+                  migration.safely_acquire_lock_for_table(:bogus_table_default) { sleep 3 }
+                end
+
+                sleep 1.1
+
+                expect(PgHaMigrations::BlockingDatabaseTransactions).to receive(:find_blocking_transactions)
+                  .at_least(3)
+                  .times
+                  .and_call_original
+
+                migration.suppress_messages do
+                  migration.safely_acquire_lock_for_table(table_name) do
+                    expect(locks_for_table(table_name, connection: alternate_connection)).not_to be_empty
```

Comment:
Should we also expect that there are no explicit locks (not sure if we distinguish) on `bogus_table_default`? If we don't have a way to distinguish I suppose we could expect that `find_blocking_transactions` returns empty before this happens?

Reply:
Addressed in https://github.com/braintree/pg_ha_migrations/pull/87/commits/bd2c1e92e2cfd7765f579f6baec38bf66f4a1a22